src/classes/Collection.cls

```diff
@@ -2,317 +2,494 @@ public with sharing class Collection {
+		return groupByIds(''+field);
 	}
 
-	public Map<Id, List<SObject>> groupByIds(Schema.SObjectField field) {
+
+	public Map<Id, List<SObject>> groupByIds(String field) {
 		return groupByIds(field, List<SObject>.class);
 	}
 
-	public Map<String, List<SObject>> groupByStrings(Schema.SObjectField field, Type listType) {
-		Map<String, List<SObject>> grouped = new Map<String, List<SObject>>();
-		for (SObject rec : records) {
-			String key = (String) rec.get(field);
-			if (!grouped.containsKey(key)) {
-				grouped.put(key, (List<SObject>) listType.newInstance());
-			}
-			grouped.get(key).add(rec);
-		}
-		return grouped;
+
+	public Map<Id, List<SObject>> groupByIds(SObjectField field, Type listType) {
+		return groupByIds(''+field, listType);
+	}
+
+
+	public Map<Id, List<SObject>> groupByIds(String field, Type listType) {
+		Map<Object, List<SObject>> result = groupByObjects(field, Id.class, listType);
+		return (Map<Id, List<SObject>>) JSON.deserialize(JSON.serialize(result), Map<Id, List<SObject>>.class);
+	}
+
+
+	public Map<String, List<SObject>> groupByStrings(SObjectField field) {
+		return groupByStrings(''+field);
 	}
 
-	public Map<String, List<SObject>> groupByStrings(Schema.SObjectField field) {
+
+	public Map<String, List<SObject>> groupByStrings(String field) {
 		return groupByStrings(field, List<SObject>.class);
 	}
 
-	public Collection pick(List<Schema.SObjectField> fields) {
-		return pick(new Set<Schema.SObjectField>(fields));
+
+	public Map<String, List<SObject>> groupByStrings(SObjectField field, Type listType) {
+		return groupByStrings(''+field, listType);
+	}
+
+
+	public Map<String, List<SObject>> groupByStrings(String field, Type listType) {
+		Map<Object, List<SObject>> result = groupByObjects(field, String.class, listType);
+		return (Map<String, List<SObject>>) JSON.deserialize(JSON.serialize(result), Map<String, List<SObject>>.class);
 	}
 
-	public Collection pick(Set<Schema.SObjectField> fields) {
-		Set<String> fieldNames = new Set<String>();
-		for (Schema.SObjectField field : fields) {
-			Schema.DescribeFieldResult describe = field.getDescribe();
-			fieldNames.add(describe.getName());
+
+	public Map<Object, List<SObject>> groupByObjects(String field, Type objectType, Type listType) {
+		Map<Object, List<SObject>> result = new Map<Object, List<SObject>>();
+
+		for(SObject record : records) {
+			Object key = JSON.deserialize(JSON.serialize(record.get(field)), objectType);
+
+			if(!result.containsKey(key)) {
+				result.put(key,(List<SObject>) listType.newInstance());
+			}
+
+			result.get(key).add(record);
 		}
-		return pick(fieldNames);
+
+		return result;
+	}
+
+
+	// PICK
+
+	public Collection pick(List<SObjectField> fields) {
+		return pick(stringList(fields));
+	}
+
+
+	public Collection pick(Set<SObjectField> fields) {
+		return pick(new List<SObjectField>(fields));
 	}
 
-	public Collection pick(Set<String> apiFieldNames) {
+	public Collection pick(Set<String> fields) {
```

Comment:
This is similar to some previous comments, we're looking to reduce the confusion. API field names are API field names, fields could field names, API field names, field names with prefixes or without them. It's not as clear.